synapse/http/server.py

```diff
@@ -217,7 +217,7 @@ def render(self, request):
         return NOT_DONE_YET
 
     @wrap_async_request_handler
-    async def _async_render_wrapper(self, request):
+    async def _async_render_wrapper(self, request: Request):
```

Comment:
trace_servlet requires a `SynapseRequest`, so I think this has to take a `SynapseRequest` too:

```suggestion
    async def _async_render_wrapper(self, request: SynapseRequest):
```